src/components/ArticlePage.jsx

```diff
@@ -0,0 +1,86 @@
+import { useState, useEffect } from "react";
+import { useParams, Link } from "react-router-dom";
+import { getArticleById, getArticleCommentsById } from "../api";
+import ProfilePicture from "./common/ProfilePicture";
+import LikeButton from "./common/LikeButton";
+import Comment from "./Comment";
+import CreateCommentForm from "./CreateCommentForm";
+import "../css/Article.css";
+import likeImg from "../assets/like.png";
+import commentImg from "../assets/chat.png";
+import { updateArticleVotes } from "../api";
+
+export default function ArticlePage() {
+  const { article_id } = useParams();
+  const [articleData, setArticleData] = useState({});
+  const { title, topic, author, body, votes, article_img_url } = articleData;
+
+  const [articleVotes, setArticleVotes] = useState(0);
+
+  const [comments, setComments] = useState([]);
+
+  useEffect(() => {
+    getArticleById(article_id).then((articleData) => {
+      setArticleData(articleData);
+      setArticleVotes(articleData.votes);
+    });
+    getArticleCommentsById(article_id).then(setComments);
+  }, []);
+
+  const updateVotes = (amount) => {
+    console.log({ amount });
+    setArticleVotes((votes) => votes + amount);
+  };
+
+  if (!articleData.article_id) {
+    return (
+      <main>
+        <h2>
+          <Link to={"/"}>Home</Link> ►
+          <Link to={`/articles/${article_id}`}>
+            {articleData.title || "..."}
+          </Link>
+        </h2>
+      </main>
+    );
+  }
+
+  return (
+    <main>
+      <h2>
+        <Link to={"/"}>Home</Link> ►
+        <Link to={`/articles/${article_id}`}>{articleData.title || "..."}</Link>
+      </h2>
+      <article>
+        <div className="article_user-profile">
+          {author ? <ProfilePicture author={author} /> : <></>}
+          <span>{author}</span>
+        </div>
+        <h3>{title}</h3>
+        <img className="article_img" src={article_img_url} />
+        <p>{body}</p>
+        <footer className="article_user-interaction">
+          <LikeButton
+            like={true}
+            updateVotes={updateVotes}
+            patchTarget={updateArticleVotes}
```

Comment:
I was a bit confused by what patchTarget was at first, but after following the trail I really like what you've done here! 😄